homeassistant/config_entries.py

```diff
@@ -1481,6 +1481,19 @@ async def async_finish_flow(
                 result["handler"], flow.unique_id
             )
 
+        if existing_entry is not None and flow.handler != "mobile_app":
+            # This causes the old entry to be removed and replaced when it
+            # should most likely update the previous entry and abort the flow
```

Comment:
Can we be clearer?
The flow should always be aborted.
In case of manual flows, integrations should implement options, reauth, reconfigure to allow the user to change settings.
In case of non user visible flows, the integration should optionally update the existing entry before aborting.

Reply:
Updated